src/analysis/construct.ml

```diff
@@ -441,8 +443,8 @@ module Gen = struct
           with Not_found ->
             let def = Env.find_type_descrs path env in
             match def with
-            | constrs, [] -> constructor env rtyp path constrs
-            | [], labels -> record env rtyp path labels
+            | Type_variant (constrs, _) -> constructor env rtyp path constrs
+            | Type_record (labels, _) -> record env rtyp path labels
             | _ -> []
```

Comment:
`| Type_abstract | Type_open -> []`
Now that there is a proper representation of all cases, better match on them :).